### PR TITLE
chore: open ledger drawer styling & error handling

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

- Style the `LedgerOpenApp` component to match the mocks in https://www.figma.com/file/VEKmwSuhnZVghkUl0VSvN2/Ledger-UX?type=design&node-id=789-420&mode=design&t=aQ5JKr0gg5tyPjyv-0
- Adds error toast when "Next" is clicked, but the correct app is not open

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Closes https://github.com/shapeshift/web/issues/6720

Low - only style changes

> What protocols, transaction types or contract interactions might be affected by this PR?

None

## Testing

Ensure `REACT_APP_FEATURE_ACCOUNT_MANAGEMENT` and `REACT_APP_FEATURE_ACCOUNT_MANAGEMENT_LEDGER` feature flags are enabled.

1. Connect a Ledger
2. Click "Manage Accounts" from the wallet menu
3. Click "Add another chain"
4. Select a Ledger supported chain using the drawer
5. See the new styling
6. Click "Next" without opening the correct Ledger app and see the error toast

### Engineering

☝️

### Operations

☝️

## Screenshots (if applicable)

Screenshot 2024-05-01 at 12 56 14 PM
Screenshot 2024-05-01 at 12 56 37 PM
Screenshot 2024-05-01 at 12 56 43 PM
Screenshot 2024-05-01 at 1 11 03 PM

### CODE REVIEW

File: src/components/AssetIcon.tsx [hunk 1]
```diff
@@ -119,3 +117,31 @@ export const AssetIcon = memo(({ assetId, showNetworkIcon, src, ...rest }: Asset
 
   return <Avatar src={src} bg={assetIconBg} icon={foxIcon} {...rest} />
 })
+
+type AssetOnLedgerProps = {
+  assetId: AssetId
+} & AvatarProps
+
+export const AssetOnLedger: React.FC<AssetOnLedgerProps> = ({ assetId, size, ...rest }) => {
```
Comment (gomesalexandre) [inline src/components/AssetIcon.tsx:125]: Not a blocker by any means but not sure how I feel about this living in this top-level shared component(s) module, would probably make more sense to colocate in Ledger/Accounts Management domain

File: src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx [hunk 2]
```diff
@@ -110,18 +113,42 @@ export const LedgerOpenApp = ({ chainId, onClose, onNext }: LedgerOpenAppProps)
     if (isValidApp) {
       onNext()
     } else {
-      console.error('Correct Ledger app is not open', slip44Key)
+      toast({
+        title: translate('walletProvider.ledger.errors.appNotOpen', { app: slip44Key }),
```
Comment (gomesalexandre) [inline src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx:117]: There's now a discrepancy between `slip44Key` for the toast and `asset.networkName` for the openApp title - do we want to consistently use `slip44Key`?

File: src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx [hunk 3]
```diff
@@ -110,18 +113,42 @@ export const LedgerOpenApp = ({ chainId, onClose, onNext }: LedgerOpenAppProps)
...
 
-  if (!asset) return null
+  const body = useMemo(() => {
+    if (!asset) return null
+    return (
+      <Center>
+        <Flex direction='column' justifyContent='center'>
+          <AssetOnLedger assetId={asset.assetId} size={'xl'} />
+          <RawText fontSize={'xl'} fontWeight={'bold'} mt={10} mb={3}>
+            {translate('accountManagement.ledgerOpenApp.title', {
```
Comment (gomesalexandre) [inline src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx:136]: i.e here
https://github.com/shapeshift/web/blob/51c41ca132294d307435abc7f7dd87aa9df2de43/src/assets/translations/en/main.json#L1810

File: src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx (hunk 2 above)
Comment (0xApotheosis) [reply src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx:117]: Ah good catch, the user doesn't care about the slip key hey. I'll update.

Comment (gomesalexandre) [reply src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx:117]: Or probably the other way around - just double checked and `keyof typeof slip44Key` are UpperCamelCased, while Ledger apps are human-readable e.g `BitcoinCash` slip44Key key (the duplicate key isn't a mistake), vs. `Bitcoin Cash` app

File: src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx [hunk 4]
```diff
@@ -110,18 +113,42 @@ export const LedgerOpenApp = ({ chainId, onClose, onNext }: LedgerOpenAppProps)
...
+              chainNamespaceDisplayName: asset?.networkName ?? '',
```
Comment (gomesalexandre) [inline src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx:137]: And as mentioned above, this won't be correct for EVM chains other than Ethereum

File: src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx (hunk 2 above)
Comment (gomesalexandre) [reply src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx:117]: Though I just read `getSlip44KeyFromChainId` again and disregard all of the above, I'm a derp - both of these will be wrong either way - `getSlip44KeyFromChainId` has the *right* SLIP44 for chains (as it should).

So what we want instead is to group EVM chains into the `ethChainId` umbrella as mentioned in the previous comment, and then use `networkName` everywhere.

See how the current Ledger `<Chains />` component handles it (notice there's only an `ethChainId` option):

https://github.com/shapeshift/web/blob/a2ffbc9def5d71bb9c7f34fa23309b350d677147/src/context/WalletProvider/Ledger/components/Chains.tsx#L77-L79

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Tested locally, conceptually looks good to me though:

- One comment WRT inconsistent app display name
- Another comment WRT EVM chains which should be using the EVM umbrella
- One potential issue with RUNE app detection

### Non-Ethereum EVM chains do not use the correct "Ethereum" app umbrella 🚫 

image
image

We'll probably want something like this:

```diff
diff --git a/src/assets/translations/en/main.json b/src/assets/translations/en/main.json
index a1dced8f4f..00e8a9b860 100644
--- a/src/assets/translations/en/main.json
+++ b/src/assets/translations/en/main.json
@@ -1807,7 +1807,7 @@
       "import": "Import"
     },
     "ledgerOpenApp": {
-      "title": "Open the %{chainNamespaceDisplayName} App",
+      "title": "Open the %{appName} App",
       "description": "To connect this chain, you will need to open the Ledger app on your device."
     }
   },
diff --git a/src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx b/src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx
index be4b47d40e..cc5607ad91 100644
--- a/src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx
+++ b/src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx
@@ -10,6 +10,7 @@ import {
   type ChainId,
   cosmosChainId,
   dogeChainId,
+  ethAssetId,
   ethChainId,
   gnosisChainId,
   ltcChainId,
@@ -17,6 +18,7 @@ import {
   polygonChainId,
   thorchainChainId,
 } from '@shapeshiftoss/caip'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
 import type { slip44Table } from '@shapeshiftoss/hdwallet-core'
 import { isLedger } from '@shapeshiftoss/hdwallet-ledger'
 import { useCallback, useEffect, useMemo } from 'react'
@@ -24,7 +26,7 @@ import { useTranslate } from 'react-polyglot'
 import { AssetOnLedger } from 'components/AssetIcon'
 import { RawText, Text } from 'components/Text'
 import { useWallet } from 'hooks/useWallet/useWallet'
-import { selectFeeAssetByChainId } from 'state/slices/selectors'
+import { selectAssetById, selectFeeAssetByChainId } from 'state/slices/selectors'
 import { useAppSelector } from 'state/store'
 
 import { DrawerContentWrapper } from './DrawerContent'
@@ -95,6 +97,12 @@ export const LedgerOpenApp = ({ chainId, onClose, onNext }: LedgerOpenAppProps)
     }
   }, [wallet, slip44Key])
 
+  const ethAsset = useAppSelector(state => selectAssetById(state, ethAssetId))
+  const appName = useMemo(() => {
+    if (isEvmChainId(chainId)) return ethAsset?.networkName
+    return asset?.networkName
+  }, [asset?.networkName, chainId, ethAsset?.networkName])
+
   useEffect(() => {
     // Poll the Ledger every second to see if the correct app is open
     const intervalId = setInterval(async () => {
@@ -114,9 +122,9 @@ export const LedgerOpenApp = ({ chainId, onClose, onNext }: LedgerOpenAppProps)
       onNext()
     } else {
       toast({
-        title: translate('walletProvider.ledger.errors.appNotOpen', { app: slip44Key }),
+        title: translate('walletProvider.ledger.errors.appNotOpen', { app: appName }),
         description: translate('walletProvider.ledger.errors.appNotOpenDescription', {
-          app: slip44Key,
+          app: appName,
         }),
         status: 'error',
         duration: 9000,
@@ -124,7 +132,7 @@ export const LedgerOpenApp = ({ chainId, onClose, onNext }: LedgerOpenAppProps)
         position: 'top-right',
       })
     }
-  }, [isCorrectAppOpen, onNext, slip44Key, toast, translate])
+  }, [isCorrectAppOpen, onNext, appName, toast, translate])
 
   const body = useMemo(() => {
     if (!asset) return null
@@ -134,7 +142,7 @@ export const LedgerOpenApp = ({ chainId, onClose, onNext }: LedgerOpenAppProps)
           <AssetOnLedger assetId={asset.assetId} size={'xl'} />
           <RawText fontSize={'xl'} fontWeight={'bold'} mt={10} mb={3}>
             {translate('accountManagement.ledgerOpenApp.title', {
-              chainNamespaceDisplayName: asset?.networkName ?? '',
+              appName,
             })}
           </RawText>
           <Text
@@ -144,7 +152,7 @@ export const LedgerOpenApp = ({ chainId, onClose, onNext }: LedgerOpenAppProps)
         </Flex>
       </Center>
     )
-  }, [asset, translate])
+  }, [appName, asset, translate])
 
   return (
     <DrawerContentWrapper
```

### Happy path - user auto-proceeds with the right app opened ✅ 


https://github.com/shapeshift/web/assets/17035424/31947fcb-bc9a-4120-b91a-614c4a2ecc1f

### Happy path - user can proceed when going from wrong app to the right app opened  ❓ 

THOR (RUNE) app is sad

https://github.com/shapeshift/web/assets/17035424/d61530f9-0723-41c6-8826-169bc90bf37c

DOGE app is happy

https://github.com/shapeshift/web/assets/17035424/637efd26-5001-41ac-9ca6-9b0762e9c9af

### Sad path - user cannot proceed with the wrong app opened ✅ 

Tested connecting Ethereum while having Litecoin app opened


https://github.com/shapeshift/web/assets/17035424/7bb58a2e-795e-43b2-8ac8-fef767ea0396

File: src/components/AssetIcon.tsx (hunk 1 above)
Comment (0xApotheosis) [reply src/components/AssetIcon.tsx]: Yeah I hear that - moved to the account management domain in https://github.com/shapeshift/web/pull/6790/commits/761ec0a6a9118087ac99aefe414df0fc007bed05.

File: src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx (hunk 2 above)
Comment (0xApotheosis) [reply src/components/ManageAccountsDrawer/components/LedgerOpenApp.tsx]: Unified EVMs under Ethereum in https://github.com/shapeshift/web/pull/6790/commits/265120d4a467506fb2c45062770e6edb78d67e01

**Comment by 0xApotheosis** [comment]
gm @gomesalexandre, ty for the excellent review.

I've unified the EVM connections/UI to use the Ethereum app in https://github.com/shapeshift/web/pull/6790/commits/265120d4a467506fb2c45062770e6edb78d67e01

We still have the THORChain connection issue, which I suspect is caused by `await this.transport.call(null, "getAppAndVersion")` returning `THORChain`, whereas the slip key is written as `Thorchain`.

This might require another HDWallet update to fix, which I'll look into tomorrow in a separate PR 🙏

**Review by gomesalexandre — APPROVED** [review]
Retested locally, LGTM now! Happy to get this in as the issue re: THOR app detection has been identified and is to be tackled in a direct follow-up 🤝 

### EVM accounts now live under the Ethereum app name umbrella


https://github.com/shapeshift/web/assets/17035424/3d2d50a9-874e-4a3c-b737-c245ef6739c4

### Non EVM app names still look sane 

https://github.com/shapeshift/web/assets/17035424/b909f798-89e0-478f-96a9-158be15ea95d

**Pull request merged**